Add flag to omit constant locations

gdemonet opened the pull request:
This script currently always generates locations for constant routes, such as `/v2`, which conflicts when we try to include multiple of those in NGINX. We add a `--omit-constants` flag to the script, so users can easily opt-out of this behaviour.

NicolasT commented:
Could you add a way to only emit the constant section, such that one doesn't need to ensure only one include file has it, and instead one can generate the constants section and include it, then for every repo, use the 'omit constants' option.

gdemonet commented:
Yes, good idea, I'll do that!

gdemonet committed:
Add flag to omit constant locations

This script currently always generates locations for constant routes,
such as `/v2`, which conflicts when we try to include multiple of those
in NGINX. We add a `--omit-constants` flag to the script, so users can
easily opt-out of this behaviour.

gdemonet committed:
Add flag to only generate constants

The counterpart of `--omit-constants`, which will be easier to use than
making sure one and only one configuration includes these.

gdemonet commented:
> Could you add a way to only emit the constant section, such that one doesn't need to ensure only one include file has it, and instead one can generate the constants section and include it, then for every repo, use the 'omit constants' option.

Done!

NicolasT commented:
Looks good now, but CI fails. I'll do a PR to fix CI, make that one land, after which this one can be rebased and should succeed then.

NicolasT commented:
CI should now be fixed in `master`. Could you merge this into this PR/branch (or rebase on top of it)?

gdemonet commented:
Yes, just rebased.
